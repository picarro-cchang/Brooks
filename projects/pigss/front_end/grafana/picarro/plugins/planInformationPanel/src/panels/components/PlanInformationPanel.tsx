import React, { Component } from "react";
import { Plan } from "./../types";
import "./planInformation.css";
import {Row, Col} from 'react-bootstrap'


interface State {
  uistatus: { [key: string]: any };
  plan: Plan;
  timeRemaining: number;
  timeStart: number;
  runType: Number;
}

export interface Props {
  uistatus: {
    [key: string]: any;
  };
  plan: Plan;
  timer: number;
  runType: Number;
}

export class PlanInformationPanel extends Component<Props, State> {
  constructor(props) {
    super(props);

    this.state = {
      uistatus: this.props.uistatus,
      plan: this.props.plan,
      timeStart: this.props.timer,
      timeRemaining: this.props.timer,
      runType: this.props.runType
    };
  }

  getBankChannelFromStep(step: number, nextStep: number) {
    const stepInfo = this.props.plan.steps[String(step)];
    const nextStepInfo = this.props.plan.steps[String(nextStep)];
    let currentStepString, nextStepString;
    if (stepInfo !== undefined){  if (stepInfo.reference != 0) {
        currentStepString = "Reference";
      } else {
        for (const bank in stepInfo.banks) {
          if (stepInfo.banks.hasOwnProperty(bank)) {
            const bank_name = this.props.plan.bank_names[bank].name;
            const bank_config = stepInfo.banks[bank];
            if (bank_config.clean != 0) {
              currentStepString = `Clean ${bank_name}`;
              break;
            } else if (bank_config.chan_mask != 0) {
              const mask = bank_config.chan_mask;
              // Find index of first set bit using bit-twiddling hack
              const channel = (mask & -mask).toString(2).length;
              const ch_name = this.props.plan.bank_names[bank].channels[channel];
              const bankNum = Number(bank);
              const portNumber = (bankNum - 1) * 8 + channel;
              currentStepString = portNumber + ": " + ch_name;
              break;
            }
          }
        }
      }}
    if (nextStepInfo !== undefined){if (nextStepInfo.reference != 0) {
        nextStepString = "Reference";
      } else {
        for (const bank in nextStepInfo.banks) {
          if (nextStepInfo.banks.hasOwnProperty(bank)) {
            const bank_name = this.props.plan.bank_names[bank].name;
            const bank_config = nextStepInfo.banks[bank];
            if (bank_config.clean != 0) {
              nextStepString = `Clean ${bank_name}`;
              break;
            } else if (bank_config.chan_mask != 0) {
              const mask = bank_config.chan_mask;
              // Find index of first set bit using bit-twiddling hack
              const channel = (mask & -mask).toString(2).length;
              const ch_name = this.props.plan.bank_names[bank].channels[channel];
              const bankNum = Number(bank);
              const portNumber = (bankNum - 1) * 8 + channel;
              nextStepString = portNumber + ": " + ch_name;
              break;
            }
          }
        }
      }}

    return { currentStepString, nextStepString };
  }

  // planFileNameUpTop() {
  //   if (
  //     this.props.uistatus.plan_loop == "ACTIVE" ||
  //     this.props.uistatus.plan_run == "ACTIVE"
  //   ) {
  //     return true;
  //   } else {
  //     return false;
  //   }
  // }

  render() {
      const currentStep = this.props.plan.current_step;
      let nextStep = 0;
      if (currentStep == this.props.plan.last_step) {
        nextStep = 1;
      } else {
        nextStep = this.props.plan.current_step + 1;
      }
      const steps = this.getBankChannelFromStep(currentStep, nextStep);
      let planName = this.props.plan.plan_filename
      if (this.props.plan.plan_filename == "") {
        planName = "No Plan Loaded"
      }
      return (
        <div className="width">
            <Row>
            <Col className={this.props.runType === 2 ? "info-row col-sm-4": "info-row col-sm-6"}>
              {this.props.runType === 0 ? (
                <div className={"col text-center br"} id="plan">
                <span className="titles">Plan Loaded: </span>
                <span className={"values"}>
                  {planName}{" "}
                </span>
              </div>
              ) : (
                <div className={"col text-center br"} id="plan">
                <span className="titles">Plan Running: </span>
                
                  {this.props.runType === 1 ? (
                  <span className={"values"}>
                    Single Port
                  </span>
                  ) : (
                  <span className={"values"}>
                    {planName}{" "}
                  </span>
                  )}
                  
              </div>
              )}
              
              </Col>
              <Col className={this.props.runType === 2 ? "text-center center-info info-row col-sm-4" : "ext-center center-info info-row col-sm-6"}>
                <div id="curr-port-line" className="row text-center">
                  <div id="curr-port" className="titles text-left">Measuring Port: </div>
                  <div id="curr-port" className="values text-left">{steps.currentStepString} </div>
                </div>
                {this.props.runType === 2 ? (
                  <div id="curr-port-time" className="row text-center">
                  <div id="curr-port" className="titles text-left"> Remaining Time: </div>
                  <div id="curr-port" className="values text-left">{this.props.timer} seconds</div>
                </div>
                ) : (null)}
              </Col>
              {this.props.runType === 2 ? (
                <Col sm={4} className={"text-center center-info-2 info-row"} id="next-port">
                <span className="titles">Next Port: </span>
                <span className="values"> {steps.nextStepString} </span>
              </Col>
              ) : (null)}
            </Row>
        </div>
      );
  }
}
