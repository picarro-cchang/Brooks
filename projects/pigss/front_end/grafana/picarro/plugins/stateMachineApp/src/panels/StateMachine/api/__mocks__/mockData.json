{
    "standby": "ACTIVE",
    "identify": "READY",
    "loop_plan": "READY",
    "run": "READY",
    "plan": "READY",
    "plan_run": "READY",
    "plan_loop": "READY",
    "reference": "READY",
    "run_plan": "READY",
    "edit": "READY",
    "clean": {
      "1": "READY",
      "3": "READY",
      "4": "READY"
    },
    "bank": {
      "1": "READY",
      "3": "READY",
      "4": "READY"
    },
    "channel": {
      "1": {
        "1": "DISABLED",
        "2": "AVAILABLE",
        "3": "DISABLED",
        "4": "AVAILABLE",
        "5": "DISABLED",
        "6": "AVAILABLE",
        "7": "DISABLED",
        "8": "AVAILABLE"
      },
      "3": {
        "1": "DISABLED",
        "2": "AVAILABLE",
        "3": "DISABLED",
        "4": "AVAILABLE",
        "5": "DISABLED",
        "6": "AVAILABLE",
        "7": "DISABLED",
        "8": "AVAILABLE"
      },
      "4": {
        "1": "DISABLED",
        "2": "AVAILABLE",
        "3": "DISABLED",
        "4": "AVAILABLE",
        "5": "DISABLED",
        "6": "AVAILABLE",
        "7": "DISABLED",
        "8": "AVAILABLE"
      }
    }
}
