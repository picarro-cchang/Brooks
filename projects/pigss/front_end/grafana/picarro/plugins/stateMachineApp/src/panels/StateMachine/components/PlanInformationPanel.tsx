import React, { Component } from 'react'
import { Plan } from './../types'

interface State {
    
}

interface Props {
    uistatus: {
        [key: string]: any;
    },
    plan: Plan
}

export class PlanInformationPanel extends Component<Props, State> {
    constructor(props) {
        super(props)

    }

    getBankChannelFromStep(step: number) {
        // const stepInfo = this.props.plan.steps[String(step)]
        // let currentStepString;
        // if (stepInfo.reference != 0) {
        //     currentStepString = 'Reference'
        // }
        // for (let bank in stepInfo.banks) {

        // }

        // return currentStepString;
    }

    render() {
        const currentStep = this.props.plan.current_step;
        const currentDuration = this.props.plan.steps[String(this.props.plan.current_step)].duration;
        const currentPort = this.getBankChannelFromStep(currentStep);
        let nextStep = 0;
        if (currentStep == this.props.plan.last_step) {
            nextStep = 1;
        } else {
            nextStep = this.props.plan.last_step + 1;
        }
        const nextPort = this.getBankChannelFromStep(nextStep);
        return (
            <div>
                <div className="plan-info">
                    Running Plan: {this.props.plan.plan_filename} Current Port: {currentStep}
                    Duration: {currentDuration} Next Port: {nextStep}
                </div>
            </div>
        )};

}
