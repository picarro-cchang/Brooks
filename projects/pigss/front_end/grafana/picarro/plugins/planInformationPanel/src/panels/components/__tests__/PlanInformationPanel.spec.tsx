import React from "react";
import { shallow, mount } from "enzyme";
import "jest-styled-components";
import WS from "jest-websocket-mock";
import { PlanInformationPanel, Props } from "./../PlanInformationPanel";
import "jest-fetch-mock";
import "react-toastify/dist/ReactToastify.css";
import mockData from "./../../api/__mocks__/mockData.json";
import mockPlan from "./../../api/__mocks__/mockPlan.json";


const defaultProps: Props = {
    uistatus: mockData,
    plan: mockPlan,
    timer: 30
};

describe("<PlanInformationPanel />", () => {
    const componentDidUpdate = jest.spyOn(PlanInformationPanel.prototype, "componentDidUpdate");
    const getBankChannelFromStep = jest.spyOn(PlanInformationPanel.prototype, "getBankChannelFromStep");
    const wrapper = mount(<PlanInformationPanel {...defaultProps}/>);
    const instance = wrapper.instance() as PlanInformationPanel;

    it("Snapshot", () => {
        expect(wrapper).toMatchSnapshot();
        console.log(wrapper.props().plan.current_step)
    });

    it("componentDidUpdate", () => {
        const plan = wrapper.props().plan
        plan.current_step = 7;
        wrapper.setProps({plan})
        expect(wrapper.props().plan.current_step).toEqual(7);
        expect(componentDidUpdate).toHaveBeenCalled();
    });

    it("getBankChannelFromStep", () => {
        const plan = wrapper.props().plan
        plan.current_step = 12;
        wrapper.setProps({plan});
        expect(getBankChannelFromStep).toHaveBeenCalled();
    });
 
});
