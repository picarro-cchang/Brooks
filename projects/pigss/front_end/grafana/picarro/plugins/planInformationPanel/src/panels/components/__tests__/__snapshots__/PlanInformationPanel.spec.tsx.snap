// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanInformationPanel /> ACTIVE Plan 1`] = `
<PlanInformationPanel
  plan={
    Object {
      "available_ports": Object {
        "1": 170,
        "2": 0,
        "3": 170,
        "4": 170,
      },
      "bank_names": Object {
        "1": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 1",
        },
        "2": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 2",
        },
        "3": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 3",
        },
        "4": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 4",
        },
      },
      "current_step": 2,
      "focus": Object {
        "column": 1,
        "row": 12,
      },
      "last_step": 12,
      "looping": false,
      "max_steps": 32,
      "num_plan_files": 2,
      "panel_to_show": 0,
      "plan_filename": "__default__",
      "plan_files": Object {
        "1": "__default__",
        "2": "tr",
      },
      "steps": Object {
        "1": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 2,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "10": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 8,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "11": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 128,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "12": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 1,
        },
        "2": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 32,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "3": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 1,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "4": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 128,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "5": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 32,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "6": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 8,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "7": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 128,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "8": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 2,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "9": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 32,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
      },
    }
  }
  timer={30}
  uistatus={
    Object {
      "bank": Object {
        "1": "READY",
        "3": "READY",
        "4": "READY",
      },
      "channel": Object {
        "1": Object {
          "1": "DISABLED",
          "2": "AVAILABLE",
          "3": "DISABLED",
          "4": "AVAILABLE",
          "5": "DISABLED",
          "6": "AVAILABLE",
          "7": "DISABLED",
          "8": "AVAILABLE",
        },
        "3": Object {
          "1": "DISABLED",
          "2": "AVAILABLE",
          "3": "DISABLED",
          "4": "AVAILABLE",
          "5": "DISABLED",
          "6": "AVAILABLE",
          "7": "DISABLED",
          "8": "AVAILABLE",
        },
        "4": Object {
          "1": "DISABLED",
          "2": "AVAILABLE",
          "3": "DISABLED",
          "4": "AVAILABLE",
          "5": "DISABLED",
          "6": "AVAILABLE",
          "7": "DISABLED",
          "8": "AVAILABLE",
        },
      },
      "clean": Object {
        "1": "READY",
        "3": "READY",
        "4": "READY",
      },
      "edit": "READY",
      "identify": "READY",
      "loop_plan": "READY",
      "plan": "READY",
      "plan_loop": "ACTIVE",
      "plan_run": "READY",
      "reference": "READY",
      "run": "READY",
      "run_plan": "READY",
      "standby": "ACTIVE",
    }
  }
>
  <div
    id="inner"
  >
    <div>
      <div
        className="row info-row"
      >
        <div
          className="col text-center br"
          id="plan"
        >
          <span
            className="titles"
          >
            Running Plan: 
          </span>
          <span
            className="values"
          >
            __default__
             
          </span>
        </div>
      </div>
      <div
        className="row info-row-2"
      >
        <div
          className="col text-center center-info info-row"
        >
          <div
            className="row text-center"
            id="curr-port-line"
          >
            <div
              className="titles text-left"
              id="curr-port"
            >
              Current Port: 
            </div>
            <div
              className="values text-left"
              id="curr-port"
            >
              6: Ch. 6
               
            </div>
          </div>
          <div
            className="row text-center"
            id="curr-port-time"
          >
            <div
              className="titles text-left"
              id="curr-port"
            >
               Remaining Time: 
            </div>
            <div
              className="values text-left"
              id="curr-port"
            >
              30
               seconds
            </div>
          </div>
        </div>
        <div
          className="col text-center center-info-2 info-row"
          id="next-port"
        >
          <span
            className="titles"
          >
            Next Port: 
          </span>
          <span
            className="values"
          >
             
            Clean Bank 1
             
          </span>
        </div>
      </div>
    </div>
  </div>
</PlanInformationPanel>
`;

exports[`<PlanInformationPanel /> Last Step = 0 1`] = `
<PlanInformationPanel
  plan={
    Object {
      "available_ports": Object {
        "1": 170,
        "2": 0,
        "3": 170,
        "4": 170,
      },
      "bank_names": Object {
        "1": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 1",
        },
        "2": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 2",
        },
        "3": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 3",
        },
        "4": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 4",
        },
      },
      "current_step": 2,
      "focus": Object {
        "column": 1,
        "row": 12,
      },
      "last_step": 0,
      "looping": false,
      "max_steps": 32,
      "num_plan_files": 2,
      "panel_to_show": 0,
      "plan_filename": "__default__",
      "plan_files": Object {
        "1": "__default__",
        "2": "tr",
      },
      "steps": Object {
        "1": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 2,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "10": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 8,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "11": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 128,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "12": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 1,
        },
        "2": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 32,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "3": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 1,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "4": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 128,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "5": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 32,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "6": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 8,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "7": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 128,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "8": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 2,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "9": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 32,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
      },
    }
  }
  timer={30}
  uistatus={
    Object {
      "bank": Object {
        "1": "READY",
        "3": "READY",
        "4": "READY",
      },
      "channel": Object {
        "1": Object {
          "1": "DISABLED",
          "2": "AVAILABLE",
          "3": "DISABLED",
          "4": "AVAILABLE",
          "5": "DISABLED",
          "6": "AVAILABLE",
          "7": "DISABLED",
          "8": "AVAILABLE",
        },
        "3": Object {
          "1": "DISABLED",
          "2": "AVAILABLE",
          "3": "DISABLED",
          "4": "AVAILABLE",
          "5": "DISABLED",
          "6": "AVAILABLE",
          "7": "DISABLED",
          "8": "AVAILABLE",
        },
        "4": Object {
          "1": "DISABLED",
          "2": "AVAILABLE",
          "3": "DISABLED",
          "4": "AVAILABLE",
          "5": "DISABLED",
          "6": "AVAILABLE",
          "7": "DISABLED",
          "8": "AVAILABLE",
        },
      },
      "clean": Object {
        "1": "READY",
        "3": "READY",
        "4": "READY",
      },
      "edit": "READY",
      "identify": "READY",
      "loop_plan": "READY",
      "plan": "READY",
      "plan_loop": "ACTIVE",
      "plan_run": "READY",
      "reference": "READY",
      "run": "READY",
      "run_plan": "READY",
      "standby": "ACTIVE",
    }
  }
>
  <div
    id="inner"
  >
    <div
      className="info-row"
      id="quick-info"
    >
      No Plan Loaded
    </div>
  </div>
</PlanInformationPanel>
`;

exports[`<PlanInformationPanel /> Snapshot 1`] = `
<PlanInformationPanel
  plan={
    Object {
      "available_ports": Object {
        "1": 170,
        "2": 0,
        "3": 170,
        "4": 170,
      },
      "bank_names": Object {
        "1": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 1",
        },
        "2": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 2",
        },
        "3": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 3",
        },
        "4": Object {
          "channels": Object {
            "1": "Ch. 1",
            "2": "Ch. 2",
            "3": "Ch. 3",
            "4": "Ch. 4",
            "5": "Ch. 5",
            "6": "Ch. 6",
            "7": "Ch. 7",
            "8": "Ch. 8",
          },
          "name": "Bank 4",
        },
      },
      "current_step": 11,
      "focus": Object {
        "column": 1,
        "row": 12,
      },
      "last_step": 12,
      "looping": false,
      "max_steps": 32,
      "num_plan_files": 2,
      "panel_to_show": 0,
      "plan_filename": "__default__",
      "plan_files": Object {
        "1": "__default__",
        "2": "tr",
      },
      "steps": Object {
        "1": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 2,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "10": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 8,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "11": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 128,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "12": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 1,
        },
        "2": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 32,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "3": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 1,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "4": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 128,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "5": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 32,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "6": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 8,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "7": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 128,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 0,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "8": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 2,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
        "9": Object {
          "banks": Object {
            "1": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "3": Object {
              "chan_mask": 0,
              "clean": 0,
            },
            "4": Object {
              "chan_mask": 32,
              "clean": 0,
            },
          },
          "duration": 20,
          "reference": 0,
        },
      },
    }
  }
  timer={30}
  uistatus={
    Object {
      "bank": Object {
        "1": "READY",
        "3": "READY",
        "4": "READY",
      },
      "channel": Object {
        "1": Object {
          "1": "DISABLED",
          "2": "AVAILABLE",
          "3": "DISABLED",
          "4": "AVAILABLE",
          "5": "DISABLED",
          "6": "AVAILABLE",
          "7": "DISABLED",
          "8": "AVAILABLE",
        },
        "3": Object {
          "1": "DISABLED",
          "2": "AVAILABLE",
          "3": "DISABLED",
          "4": "AVAILABLE",
          "5": "DISABLED",
          "6": "AVAILABLE",
          "7": "DISABLED",
          "8": "AVAILABLE",
        },
        "4": Object {
          "1": "DISABLED",
          "2": "AVAILABLE",
          "3": "DISABLED",
          "4": "AVAILABLE",
          "5": "DISABLED",
          "6": "AVAILABLE",
          "7": "DISABLED",
          "8": "AVAILABLE",
        },
      },
      "clean": Object {
        "1": "READY",
        "3": "READY",
        "4": "READY",
      },
      "edit": "READY",
      "identify": "READY",
      "loop_plan": "READY",
      "plan": "READY",
      "plan_loop": "READY",
      "plan_run": "READY",
      "reference": "READY",
      "run": "READY",
      "run_plan": "READY",
      "standby": "ACTIVE",
    }
  }
>
  <div
    id="inner"
  >
    <div>
      <div
        className="info-row"
        id="quick-info"
      >
        Loaded Plan: 
        __default__
      </div>
    </div>
  </div>
</PlanInformationPanel>
`;
